welcome.js

```diff
@@ -0,0 +1,45 @@
+
+
+class WelcomeModal {
+  constructor(options) {
+    this.languages = options.languages
+    this.onLanguageSelect = options.onLanguageSelect || (() => {})
+
+    this.el = document.createElement('div')
+    this.el.className = 'modal-wrap'
+    this.el.innerHTML = this.render(this.languages)
+    this.langButtons = this.el.querySelectorAll('.language-button')
+
+    Array.prototype.slice.call(this.langButtons).forEach(b => {
+      b.addEventListener('click', evt => {
+        evt.preventDefault()
+        const lang = b.value
+        this.onLanguageSelect(lang)
+        this.close()
+      })
+    })
+  }
+
+  render(languages) {
+
+    const buttons = languages.map(l => `<button class="language-button" value="${l.name}">${l.label}</button>`).join('')
+
+    return `
+      <div class="modal">
```

Comment:
Would it be possible to make it clear that this is a modal / hint at the content behind the modal by adding margin around the edges & not making the modal 100% height?

Reply:
Good idea

Reply:
Yeah good idea, thanks!